snapcraft/tests/fixture_setup.py

```diff
@@ -1013,3 +1013,23 @@ def _stop_fake_server(self, thread):
         self.server.shutdown()
         self.server.socket.close()
         thread.join()
+
+
+class SharedCache(fixtures.Fixture):
+
+    def __init__(self, name):
+        super().__init__()
+        self.name = name
+
+    def setUp(self):
+        super().setUp()
+        shared_cache_dir = os.path.join(
+            tempfile.gettempdir(), 'snapcraft_test_cache_{}'.format(self.name))
```

Comment:
This makes it so that there is no share cache

Reply:
Oh, my mistake, just saw what `tempfile.gettempdir` is doing. Why not set it the the actual `xdg_cache_home`?

Reply:
The xdg_cache_home is patched in all the tests. I started complicating things trying to make an exception to that rule only for ros on the other PR. So, I jumped back a little and thought that we don't really want to pollute home in any test. This sounded like a clean and better approach to me.